packages/origin-backend/src/pods/organization/organization.controller.ts

```diff
@@ -16,32 +20,71 @@ import {
     BadRequestException,
     UnprocessableEntityException,
     Delete,
-    Put
+    Put,
+    UseGuards,
+    Query
 } from '@nestjs/common';
 import { InjectRepository } from '@nestjs/typeorm';
+import { AuthGuard } from '@nestjs/passport';
 
 import { Organization } from './organization.entity';
 import { StorageErrors } from '../../enums/StorageErrors';
+import { UserDecorator } from '../user/user.decorator';
+import { UserService } from '../user/user.service';
+import { OrganizationInvitation } from './organizationInvitation.entity';
+import { User } from '../user/user.entity';
 
 @Controller('/Organization')
 export class OrganizationController {
     constructor(
         @InjectRepository(Organization)
-        private readonly organizationRepository: Repository<Organization>
+        private readonly organizationRepository: Repository<Organization>,
+        @InjectRepository(OrganizationInvitation)
+        private readonly organizationInvitationRepository: Repository<OrganizationInvitation>,
+        private readonly userService: UserService
     ) {}
 
     @Get()
     async getAll() {
-        console.log(`<GET> Organization all`);
-
         return this.organizationRepository.find();
     }
 
+    @Get('invitation')
+    @UseGuards(AuthGuard('jwt'))
+    async getInvitations(
+        @UserDecorator() loggedUser: IUser,
+        @Query('email') emailFromQuery: string,
+        @Query('organization') organizationId: string
+    ): Promise<IOrganizationInvitation[]> {
+        const findConditions: FindConditions<OrganizationInvitation> = {};
+
+        if (emailFromQuery) {
+            if (emailFromQuery === loggedUser.email) {
+                findConditions.email = loggedUser.email;
+            } else {
+                throw new BadRequestException(
+                    `You can only get invitations received to logged user email.`
+                );
+            }
+        }
+
+        if (organizationId) {
+            findConditions.organization = { id: parseInt(organizationId, 10) };
+        }
+
+        return (this.organizationInvitationRepository.find({
+            where: findConditions,
+            loadRelationIds: true
+        }) as Promise<Omit<IOrganizationInvitation, 'organization'>[]>) as Promise<
+            IOrganizationInvitation[]
+        >;
+    }
+
     @Get('/:id')
     async get(@Param('id') id: string) {
-        console.log(`<GET> Organization/${id}`);
-
-        const existingEntity = await this.organizationRepository.findOne(id);
+        const existingEntity = await this.organizationRepository.findOne(id, {
+            loadRelationIds: true
+        });
 
         if (!existingEntity) {
```

Comment:
You can replace that with interceptor like:

```
import {
    Injectable,
    NestInterceptor,
    ExecutionContext,
    CallHandler,
    NotFoundException
} from '@nestjs/common';
import { Observable } from 'rxjs';
import { map } from 'rxjs/operators';

@Injectable()
export class EmptyResultInterceptor implements NestInterceptor {
    intercept(context: ExecutionContext, next: CallHandler): Observable<any> {
        return next.handle().pipe(
            map(data => {
                if (!data) {
                    throw new NotFoundException();
                }
                return data;
            })
        );
    }
}
```

Reply:
This would affect a major part of codebase, I'd prefer not to do it as part of this pull request which already contains big changes.